File: Bai_66/main.cpp
#include<iostream>
#include<map>
using namespace std;
void prime_factorization(long long n){
    map<long long, int> factors;
    long long i = 2;
    while(i * i <= n){
        if(n % i == 0){
            int cnt = 0;
            while(n % i == 0){
                n /= i;
                cnt += 1;
            }
            factors[i] = cnt;
        }
        i += 1;
    }
    if(n > 1) factors[n] = 1;
    for(auto factor: factors)
        cout << factor.first << "^" << factor.second << " ";
    cout << endl;
}

int main(){
    long long n;
    do{
        cin >> n;
        if(n == 0) break;
        prime_factorization(n);
    } while (n != 0);
    return 0;
}

File: Bai_66/main.py
def prime_factorization(n):
    factors = []
    i = 2
    while i * i <= n:
        if n % i == 0:
            cnt = 0
            while n % i == 0:
                n //= i
                cnt += 1
            factors.append((i, cnt))
        i += 1
    if n > 1:
        factors.append((n, 1))

    for i in range(len(factors)):
        print(factors[i][0], "^", factors[i][1], end="")
        if i < len(factors) - 1:
            print(" ", end="")
    print()

while True:
    n = int(input())
    if n == 0:
        break
    prime_factorization(n)
